jdbc/src/main/java/org/apache/iotdb/jdbc/IoTDBQueryResultSet.java

```diff
@@ -1251,15 +1294,14 @@ private String getValueByName(String columnName) throws SQLException {
     if (columnName.equals(TIMESTAMP_STR)) {
       return String.valueOf(record.getTimestamp());
     }
-    int tmp = columnInfoMap.get(columnName);
-    int i = 0;
-    for (Field field : record.getFields()) {
-      i++;
-      if (i == tmp - 1) {
-        return field.getDataType() == null ? null : field.getStringValue();
-      }
+    int index = columnInfoMap.get(columnName) - START_INDEX;
+    Field field = record.getFields().get(index);
```

Comment:
What if the index is out of bound? 
In the old code, no `i == tmp - 1` so null is returned.
Besides, will field ever be null?

Reply:
> What if the index is out of bound? In the old code, if no `i == tmp - 1`, `null` is returned.
> Besides, will field ever possible be assigned null in this statement?
The point you mentioned is exactly a problem that need be fixed. Thanks.